pkg/sbom/scanner/scanner_test.go

```diff
@@ -0,0 +1,245 @@
+	"time"
+
+	compConfig "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/sbom"
+	"github.com/DataDog/datadog-agent/pkg/sbom/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+
+	cyclonedxgo "github.com/CycloneDX/cyclonedx-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type scanRequest struct {
+	collectorName   string
+	scanRequestType string
+	id              string
+}
+
+// Collector returns the collector name
+func (s *scanRequest) Collector() string {
+	return s.collectorName
+}
+
+// Type returns the scan request type
+func (s *scanRequest) Type() string {
+	return s.scanRequestType
+}
+
+// ID returns the scan request ID
+func (s *scanRequest) ID() string {
+	return s.id
+}
+
+var _ sbom.ScanRequest = (*scanRequest)(nil)
+
+type mockReport struct {
+	id string
+}
+
+// ToCycloneDX returns a mock BOM
+func (m mockReport) ToCycloneDX() (*cyclonedxgo.BOM, error) {
+	return &cyclonedxgo.BOM{}, nil
+}
+
+// ID returns the report ID
+func (m mockReport) ID() string {
+	return m.id
+}
+
+var _ sbom.Report = mockReport{}
+
+// Test retry handling in case of an error
+func TestRetryLogic_Error(t *testing.T) {
+	// Create a workload meta global store
+	workloadmetaStore := fxutil.Test[workloadmeta.Mock](t, fx.Options(
+		logimpl.MockModule(),
+		compConfig.MockModule(),
+		fx.Supply(context.Background()),
+		fx.Supply(workloadmeta.NewParams()),
+		workloadmeta.MockModuleV2(),
+	))
+	workloadmeta.SetGlobalStore(workloadmetaStore)
+	defer workloadmeta.SetGlobalStore(nil)
+
+	// Store the image
+	imageID := "id"
+	workloadmetaStore.Set(&workloadmeta.ContainerImageMetadata{
+		EntityID: workloadmeta.EntityID{
+			ID:   imageID,
+			Kind: workloadmeta.KindContainerImageMetadata,
+		},
+	})
+
+	// Create a mock collector
+	collName := "mock"
+	mockCollector := collectors.NewMockCollector()
+	resultCh := make(chan sbom.ScanResult, 1)
+	errorResult := sbom.ScanResult{Error: errors.New("scan error")}
+	expectedResult := sbom.ScanResult{Report: mockReport{id: imageID}}
+	mockCollector.On("Options").Return(sbom.ScanOptions{})
+	mockCollector.On("Scan", mock.Anything, mock.Anything).Return(errorResult).Twice()
+	mockCollector.On("Scan", mock.Anything, mock.Anything).Return(expectedResult).Once()
+	mockCollector.On("Channel").Return(resultCh)
+	mockCollector.On("Shutdown")
+	mockCollector.On("Type").Return(collectors.ContainerImageScanType)
+	collectors.RegisterCollector(collName, mockCollector)
+
+	// Set up the configuration
+	cfg := config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	cfg.Set("sbom.scan_queue.base_backoff", "2s", model.SourceAgentRuntime)
+	cfg.Set("sbom.scan_queue.max_backoff", "5s", model.SourceAgentRuntime)
```

Comment:
I think it would be good to document why we need to override the default values here.
The test would be too slow if we didn't.